### ISSUE TITLE
Use consistent terminology describing nearby cells

### ISSUE THREAD

**Issue by abalkin** [ipython/ipython#7807]
The keyboard shortcuts screen uses three different pair of terms to describe the two cell adjacent to the current:

1. below/above
1. next/previous
1. before/after

For example, shift-enter is described as "run cell, select **below**", "J" is "select **next** cell", and "B" is "insert cell *after*".  The last choice is truly confusing: "B" means "**a**fter" and "A" means "**b**efore".

**Comment by takluyver** [comment]
Ha, yes. At least for the `a` and `b` shortcuts we should certainly use the terms 'above' and 'below' to help people remember them.

@Carreau this appears to tie into the action names - are we auto generating help strings from action names now? That seems less than ideal, but that's another question. Can you look into this? I think at a minimum we should get rid of 'before' and 'after' in the naming because of the potential confusion with the shortcuts for 'above' and 'below'.

**Comment by Carreau** [comment]
> are we auto generating help strings from action names now

Yes if no help string is given. But action names are even less stable that JS API :-P 

I'm not sure above an below are the good one as notebook presentation can be non-vertical, like in reveal. Consistency would be great, but "A Foolish Consistency is the Hobgoblin of Little Minds".
And where above/below might make sens in some place (UI) before/after make sens (in API)
so I would be careful in choosing.

ping @ellisonbg that will be interested.

**Comment by abalkin** [comment]
@Carreau - "a" and "b" shortcuts don't make sense in reveal because slides are not editable.  In fact, "b" is a shortcut to "blackout" there.  I think when you are editing the the notebook, cells are always arranged vertically.

If some alternative application uses a different arrangement - it should change the shortcuts.

**Comment by takluyver** [comment]
Honestly I don't think it's a big problem which names we use. It's pretty obvious what they all mean, and I doubt anyone could really miss the relationship between spatial organisation (above/below) and logical sequence (next/previous).

But the help text for the 'a' and 'b' shortcuts really shouldn't call them 'before' and 'after'. That's just needlessly confusing. Marked for 3.0 so we can at least fix that.

@abalkin well, you can edit notebooks in reveal mode using https://github.com/damianavila/live_reveal ;-)

**Comment by Carreau** [comment]
> "a" and "b" shortcuts don't make sense in reveal because slides are not editable

Yes they are with `live_reveal`, but in  a slide true, they would be vertically arranged.
I just want to point out that the above/below might not always make sens. 

I had a prototype of notebook using [gridster](http://gridster.net/#intro), where above/below was not right. 
But agreed this is extreme case. 

Agreed that before/after _**in User Interface**_ is confusing, and that above/below match the a/b of shortcut though.

**Comment by abalkin** [comment]
Here is the screenshot after 38a2f4a:

![screen shot 2015-02-17 at 9 22 41 pm](https://cloud.githubusercontent.com/assets/535197/6241464/9ea51d0c-b6eb-11e4-90cf-50280d771d74.png)

Should I change J/K and up/down-error help as well?

**Comment by Carreau** [comment]
Sure, I guess that would be fine. Example notebook might need some checking also.

**Comment by takluyver** [comment]
Someone might want to argue that with the current state (after @abalkin's commit), above/below refer to positions between cells, while next/previous refer to adjacent cells. That has a kind of consistency.

I don't have a strong feeling either way, but it's worth considering.

**Comment by abalkin** [comment]
I would also bring "D,D" entry next to "I',|" and "O,O".  Is "," really necessary in those?  It is mildly confusing.  Why not just "DD"?

**Comment by Carreau** [comment]
> I would also bring "D,D" entry next to "I',|" and "O,O". Is "," really necessary in those? It is mildly confusing. Why not just "DD"?

The logic that make the keymap is generic. you can bind to `Ctrl-X,Meta-C,Meta-b,u,t,t,e,r,f,l,y` it will understand it. to `Ctrl-X,Meta-C,Meta-b,Meta-u,Meta-t,Meta-t,Meta-e,Meta-r,Meta-f,Meta-l,Meta-y` is different, but would look weird without commas.

If you think of  a better logic that would be great.

**Comment by abalkin** [comment]
Do you also have generic logic that uppercases letter shortcuts?

Emacs uses conventions like this:

```
Get help           C-h  (Hold down CTRL and press h)
Emacs manual       C-h r        Browse manuals     C-h i
Emacs tutorial     C-h t        Undo changes       C-x u
Buy manuals        C-h RET      Exit Emacs         C-x C-c
```

So your butterfly command will be `Ctrl-X Meta-C Meta-b u t t e r f l y`.

Vim help omits spaces when command only consists of letters.

Can you point me to the code implementing current logic?  I'll try to come up with something, but probably under a separate issue.

**Comment by Carreau** [comment]
Upper casing and symbol are dine only on OS X. Should be in quickhelp.js IIRC. 

Envoyé de mon iPhone

**Comment by minrk** [comment]
Should this have been closed by #7811?

**Comment by Carreau** [comment]
Not completelty IIRC.

**Comment by abalkin** [comment]
I think we can still improve the spelling for two key presses shortcuts as a part of this issue.
